File: 68.py
def lcaDeepestLeaves(root):
    def dfs(node):
        if not node:
            return (0, None)
        l_depth, l_node = dfs(node.left)
        r_depth, r_node = dfs(node.right)
        if l_depth > r_depth:
            return (l_depth + 1, l_node)
        elif r_depth > l_depth:
            return (r_depth + 1, r_node)
        else:
            return (l_depth + 1, node)
    return dfs(root)[1]

File: 68.js
function lcaDeepestLeaves(root) {
    function dfs(node) {
        if (!node) return [0, null];
        const [ld, ln] = dfs(node.left);
        const [rd, rn] = dfs(node.right);
        if (ld > rd) return [ld + 1, ln];
        else if (rd > ld) return [rd + 1, rn];
        else return [ld + 1, node];
    }
    return dfs(root)[1];
}
